### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by shaaibu7**
**Completed Project Requirement**

- [ ] Splash screen
   
       A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- [ ] Sign up and log in pages

       The user should be able to register in the app with full name, email and password (all mandatory).
      The user can log into the app using email and password.
      If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ] Home page (categories page)

        When the user logs in, they are presented with the categories page.
        For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that 
        category.
        When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
        There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [ ] Transactions page

        For a given category, the list of transactions is presented, ordered by the most recent.
        At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in 
        that category).
        There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
        When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new category" page

          The user fills out a form to create a new category, indicating their name and icon (both mandatory).
          The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
          When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new transaction" page

          The user fills out a form to create a new transaction with:
                name (mandatory)
                amount (mandatory)
                categories (mandatory at least one)
          The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that 
           category.
          When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- [ ] Create unit and integration tests for all the components.
- [ ] Project deployment.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@
 
 ## 📹 Video Description
 
-- [Video Description Link](https://www.loom.com/share/bb51379e56854720a0178d1cf8c4d2c5?sid=e109121e-20f7-49f7-b0e0-9bfd8f54962d)
+- [Video Description](https://www.loom.com/share/bb51379e56854720a0178d1cf8c4d2c5?sid=e109121e-20f7-49f7-b0e0-9bfd8f54962d)
```
Comment (Davidosky007) [inline README.md:70]: * Kindly ensure your README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: README.md [hunk 2]
```diff
@@ -67,7 +67,7 @@
...
```
Comment (Davidosky007) [inline README.md:71]: * I get noticed this bug when I try to submit an empty or half-filled form the form submits anyways with an error kindly look into it 

https://github.com/shaaibu7/Budget-app/assets/52497246/e6b312b4-fe2e-48cc-96a7-18ff53d09711

File: README.md [hunk 3]
```diff
@@ -67,7 +67,7 @@
...
```
Comment (Davidosky007) [inline README.md:72]: * Kindly fix the failing test i get when i run `rspec` 
image

File: app/views/groups/new.html.erb [hunk 4]
```diff
@@ -15,7 +15,7 @@
             <a href="/">
                 <i class="fa fa-arrow-left fa-2x text-white" aria-hidden="true"></i>
             </a>
-            <h2 class="m-0 p-0">Create Category</h2>
+            <h2 class="m-0 p-0">Create New Category</h2>
```
Comment (Davidosky007) [inline app/views/groups/new.html.erb:18]: * Kindly stick to the design given as much as possible as it currently your :

> * Colors
> * Typography: font face, size and weight.
> * Layout: composition and space between elements. dont match the given design 
image
image
image
* Also ensure you handle n +1 issues

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @shaaibu7 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Professional video presentation 👍 
* No linter errors 👍 
* Live demo link works 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (peterrobert) [reply README.md:70]: This change has been done well

File: README.md (hunk 2 above)
Comment (peterrobert) [reply README.md:71]: This change has been done well

File: README.md (hunk 3 above)
Comment (peterrobert) [reply README.md:72]: This change has been done well

File: app/views/groups/new.html.erb (hunk 4 above)
Comment (peterrobert) [reply app/views/groups/new.html.erb:18]: Great work. Good work solving the n+1 issues also

**Review by peterrobert — APPROVED** [review]
Hi @shaaibu7, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**